### PR TITLE
Preliminary support for Extreme/Brocade SLXOS

### PR DESCRIPTION
**Pull request by dcorbe**
This patch adds preliminary support for Extreme/Brocade SLXOS systems.  

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by dcorbe** [comment]
I'm not sure why travis is complaining.   The YAML passes validation in my IDE and there's no additional PHP or Python added.

**Comment by Jellyfrog** [comment]
> I'm not sure why travis is complaining. The YAML passes validation in my IDE and there's no additional PHP or Python added.

Need to add tests, see comment above

**Comment by dcorbe** [comment]
How important is it that new submissions pass the codeclimate tests?   Because it was mostly copy/pasted from the Brocade/NOS stuff.   This is an "NOS-like" device, but it is NOT Brocade's NOS.

**Comment by Jellyfrog** [comment]
Just see it as tips. You can ignore It

File: includes/polling/os/slxos.inc.php [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?php
```
Comment (Jellyfrog) [inline includes/polling/os/slxos.inc.php:1]: Could you please move this file to the new format, see https://docs.librenms.org/Developing/os/Initial-Detection/#os-discovery for more info, sorry for the inconvenience

Comment (Jellyfrog) [inline includes/polling/os/slxos.inc.php:1]: Could you please move this file to the new format, see https://docs.librenms.org/Developing/os/Initial-Detection/#os-discovery for more info, sorry for the inconvenience

**Review by Jellyfrog — CHANGES_REQUESTED** [review]

**Comment by bakerds** [comment]
Thanks @dcorbe for this work. It was on my to-do list as well :)

In testing, I see this during processor discovery:
`MIB search path: /opt/librenms/mibs:/opt/librenms/mibs/extreme
Cannot find module (SW-MIB): At line 1 in (none)
swCpuUsage: Unknown Object Identifier (Sub-id not found: (top) -> swCpuUsage)`

Did you duplicate SW-MIB from mibs/brocade directory into mibs/extreme on your system, and did you mean to include that file in this PR?

**Comment by bakerds** [comment]
Looks like there are some dependencies on older Brocade MIBs, which I think should be copied from mibs/brocade to mibs/extreme:
- SYSTEM-MIB
- SWBASE-MIB
- Brocade-TC-MIB
- SW-MIB

@Jellyfrog's request can be implemented by:
- delete includes/polling/os/slxos.inc.php
- add these lines to includes/definitions/discovery/slxos.yaml:
```
mib: SW-MIB:SYSTEM-MIB
modules:
    os:
        hardware: ENTITY-MIB::entPhysicalDescr.1
        serial: ENTITY-MIB::entPhysicalSerialNum.1
        version: SYSTEM-MIB::swFirmwareVersion.0
```

**Comment by dcorbe** [comment]
> Thanks @dcorbe for this work. It was on my to-do list as well :)
> 
> In testing, I see this during processor discovery:
> `MIB search path: /opt/librenms/mibs:/opt/librenms/mibs/extreme`
> `Cannot find module (SW-MIB): At line 1 in (none)`
> `swCpuUsage: Unknown Object Identifier (Sub-id not found: (top) -> swCpuUsage)`
> 
> Did you duplicate SW-MIB from mibs/brocade directory into mibs/extreme on your system, and did you mean to include that file in this PR?

The tests are passing on my system.   What command did you run to get this error message?

File: includes/polling/os/slxos.inc.php (hunk 1 above)
Comment (dcorbe) [reply includes/polling/os/slxos.inc.php]: Done.   Sorry for the delay.

**Comment by Jellyfrog** [comment]
Need to run step 2 of the data gathering again; https://docs.librenms.org/Developing/os/Test-Units

**Comment by bakerds** [comment]
> > Thanks @dcorbe for this work. It was on my to-do list as well :)
> > In testing, I see this during processor discovery:
> > `MIB search path: /opt/librenms/mibs:/opt/librenms/mibs/extreme`
> > `Cannot find module (SW-MIB): At line 1 in (none)`
> > `swCpuUsage: Unknown Object Identifier (Sub-id not found: (top) -> swCpuUsage)`
> > Did you duplicate SW-MIB from mibs/brocade directory into mibs/extreme on your system, and did you mean to include that file in this PR?
> 
> @bakerds The tests are passing on my system. What command did you run to get this error message?

During discovery.php run.
All good since your last commit, though. :heavy_check_mark:

**Comment by dcorbe** [comment]
> Need to run step 2 of the data gathering again; https://docs.librenms.org/Developing/os/Test-Units

@Jellyfrog Just did, sorry about that.

**Comment by dcorbe** [comment]
What's the over/under on getting this merged?

**Comment by murrant** [comment]
Any reason the sensors are not done in yaml?  They look they could be done in yaml to simplify and futureproof them.

**Comment by dcorbe** [comment]
> Any reason the sensors are not done in yaml? They look they could be done in yaml to simplify and futureproof them.

I'll work on that today.

**Comment by murrant** [comment]
@dcorbe You will want to use skip_values to filter only the entries you want for each type.

**Comment by kaniini** [comment]
can somebody elaborate on how the sensors-to-YAML conversion should be done?  I can do it this weekend.

**Pull request closed without merging**